<?xml version="1.0"?>
<launch>

  <!-- Arguments -->
  <arg name="map_file" default="$(find rapid)/maps/0201test01.yaml"/>
  <!--arg name="map_file" default="/home/smarthc/mapmap.yaml"/-->
  <arg name="open_rviz" default="true"/>
  <arg name="move_forward_only" default="false"/>

  <!-- minirapid1 model -->
  <include file="$(find rapid)/launch/bringup/minirapid_bringup.launch"/>
  <node name="tf_footprint_base" pkg="tf"  type="static_transform_publisher" args="0 0 -0.03 0 0 0 base_link base_footprint 40" />


  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>


  <!-- odom_combined -->
  <!--include file="$(find rapid)/launch/navigation/robot_pose_ekf.launch" />
  <remap from="robot_pose_ekf/odom_combined" to="odom" /-->

   <!--localization -->
   <!--node name="cartographer_node" pkg="cartographer_ros" type="cartographer_node" args="
	-configuration_directory
	   $(find rapid)/cartographer_ros/configuration_files
        -configuration_basename rapid_local_only_mode.lua" output="screen">
   </node-->

   <include file="$(find rapid)/launch/navigation/amcl_odom.launch"/>
 
  <!-- move_base -->
  <include file="$(find rapid)/launch/navigation/move_base_global-planner.launch">
    <!--arg name="cmd_vel_topic" value="/my_cmd_vel" /-->
    <arg name="move_forward_only" value="$(arg move_forward_only)"/>
  </include>

  <!-- rviz -->
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find rapid)/rviz/minirapid_navigation.rviz"/>
  </group>
</launch>
